components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1464,53 +1463,49 @@ void RewardsServiceImpl::SetConfirmationsIsReady(const bool is_ready) {
     ads_service->SetConfirmationsIsReady(is_ready);
 }
 
-void RewardsServiceImpl::GetAdsNotificationsHistory() {
+void RewardsServiceImpl::ConfirmationsTransactionHistoryDidChange() {
+    for (auto& observer : observers_)
+    observer.OnConfirmationsHistoryChanged(this);
+}
+
+void RewardsServiceImpl::GetAdsNotificationsHistory(
+    brave_rewards::AdsNotificationsHistoryCallback callback) {
   if (!Connected()) {
     return;
   }
 
   auto earnings_range = GetEarningsRange();
 
-  bat_ledger_->GetAdsNotificationsHistory(earnings_range.first,
-      earnings_range.second, base::BindOnce(
-      &RewardsServiceImpl::OnGetAdsNotificationsHistoryMojoProxy, AsWeakPtr()));
-}
-
-void RewardsServiceImpl::ConfirmationsTransactionHistoryDidChange() {
+  bat_ledger_->GetAdsNotificationsHistory(
+      earnings_range.first,
+      earnings_range.second,
+      base::BindOnce(&RewardsServiceImpl::OnGetAdsNotificationsHistoryMojoProxy,
```

Comment:
nit: see comment below, can we drop the `MojoProxy` on this? It doesn't make any sense here

Reply:
done